File: Glucloser/app/src/main/java/com/nlefler/glucloser/models/CheckInPushedData.java
package com.nlefler.glucloser.models;

/**
 * Created by Nathan Lefler on 12/29/14.
 */
public class CheckInPushedData {
    private String alert;
    private CheckInData checkInData;

    static class CheckInData {
        private String venueId;
        private String venueName;
        private String venueLat;
        private String venueLon;

        public CheckInData() {

        }
    }

    public CheckInPushedData() {

    }

    public String getVenueId() {
        return checkInData.venueId;
    }

    public String getVenueName() {
        return checkInData.venueName;
    }

    public float getVenueLat() {
        if (checkInData.venueLat != null && !checkInData.venueLat.isEmpty()) {
            return Float.valueOf(checkInData.venueLat);
        }
        else {
            return 0;
        }
    }

    public float getVenueLon() {
        if (checkInData.venueLon != null && !checkInData.venueLon.isEmpty()) {
            return Float.valueOf(checkInData.venueLon);
        }
        else {
            return 0;
        }
    }

}

File: Glucloser/app/src/main/java/com/nlefler/glucloser/models/CheckInPushedData.kt
package com.nlefler.glucloser.models

/**
 * Created by Nathan Lefler on 12/29/14.
 */
public class CheckInPushedData {
    private val checkInData: CheckInData? = null

    class CheckInData {
        internal val venueId: String? = null
        internal val venueName: String? = null
        internal val venueLat: String? = null
        internal val venueLon: String? = null
    }

    public fun getVenueId(): String {
        return checkInData?.venueId ?: ""
    }

    public fun getVenueName(): String {
        return checkInData?.venueName ?: ""
    }

    public fun getVenueLat(): Float {
        if (checkInData?.venueLat?.length() ?: 0 > 0) {
            return java.lang.Float.valueOf(checkInData!!.venueLat)
        } else {
            return 0f
        }
    }

    public fun getVenueLon(): Float {
        if (checkInData?.venueLon?.length() ?: 0 > 0) {
            return java.lang.Float.valueOf(checkInData!!.venueLon)
        } else {
            return 0f
        }
    }

}
